### PR TITLE
Feature/489 bri 3 accounts model

### PR DESCRIPTION
**Pull request by skosito**
# Description

According to document referenced in linked issue https://github.com/eea-oasis/baseline/issues/489. This covers first task, adding simple model, more implementation will come through CRUD task following up soon.

## Related Issue

https://github.com/eea-oasis/baseline/issues/489

## Motivation and Context

Part of BRI-3 (simple reference implementation) effort.

## How Has This Been Tested

Will be tested with CRUD task, for now it is just simple model.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

File: examples/bri-3/components/identity/bpiAccount/bpiAccount.ts
Comment (skosito) [inline examples/bri-3/components/identity/bpiAccount/bpiAccount.ts:9]: we will probably add better way for nonce initialization, maybe some function that increments it, but for now maybe its enough to have it in constructor?

Comment (fosgate29) [reply examples/bri-3/components/identity/bpiAccount/bpiAccount.ts:9]: It is enough to have it in constructor I believe

**Review by fosgate29 — DISMISSED** [review]
lgtm

File: examples/bri-3/components/identity/bpiAccount/bpiAccount.interface.ts
Comment (Therecanbeonlyone1969) [inline examples/bri-3/components/identity/bpiAccount/bpiAccount.interface.ts:4]: that should be an array in order to allow for flexibility and extensibility. Plus typically, a BPI account will have two owners since it connects to a joint state.

File: examples/bri-3/components/identity/bpiAccount/bpiAccount.ts
Comment (Therecanbeonlyone1969) [inline examples/bri-3/components/identity/bpiAccount/bpiAccount.ts:6]: see previous comment

File: examples/bri-3/components/identity/bpiAccount/bpiAccount.ts
Comment (Therecanbeonlyone1969) [inline examples/bri-3/components/identity/bpiAccount/bpiAccount.ts:5]: since there is more than one way to generate a deterministic nonce a string might be more appropriate

**Review by Therecanbeonlyone1969 — CHANGES_REQUESTED** [review]
See comments

File: examples/bri-3/components/identity/bpiAccount/bpiAccount.interface.ts
Comment (ognjenkurtic) [reply examples/bri-3/components/identity/bpiAccount/bpiAccount.interface.ts:4]: Could you please elaborate on the second sentence. Why should it have two owners typically? Isn't it more common that org A would work with one BpiAccount and org B with another?

Comment (ognjenkurtic) [reply examples/bri-3/components/identity/bpiAccount/bpiAccount.interface.ts:4]: I agree that it should be an array, multiple owners makes sense in multisig scenarios.

**Review by ognjenkurtic — COMMENTED** [review]
I am ok with the PR as soon as @Therecanbeonlyone1969 regarding multiple owners is addresed

File: examples/bri-3/components/identity/bpiAccount/bpiAccount.interface.ts
Comment (Kasshern) [reply examples/bri-3/components/identity/bpiAccount/bpiAccount.interface.ts:4]: Perhaps we should we should discuss and conclude a scope of functionality for alignment of team and maintainers in future PRs. Originally the team agreed on the "simplest implementation" to cover our intended use case. We seem to ever so slowly be drifting towards accommodating common use case needs that fall outside the "MUST" requirements of the standard; a direction in which has my support.

**Review by Kasshern — COMMENTED** [review]
I concur with previous comments and approve pending suggested changes of @Therecanbeonlyone1969

File: examples/bri-3/components/identity/bpiAccount/bpiAccount.interface.ts
Comment (Therecanbeonlyone1969) [reply examples/bri-3/components/identity/bpiAccount/bpiAccount.interface.ts:4]: If we want the SRI to be useful for multi party coordination WITHOUT being clunky aka you need to double or triple the BPI accounts. And will impact adoption. 

I agree we should keep it as simple as possible while ensure good usability.

File: examples/bri-3/components/identity/bpiAccount/bpiAccount.ts
Comment (Manik-Jain) [reply examples/bri-3/components/identity/bpiAccount/bpiAccount.ts:5]: Considering the upper limit of TS number type, I think we shall implement it as bignumber right since the beginning.

File: examples/bri-3/components/identity/bpiAccount/bpiAccount.ts
Comment (Manik-Jain) [reply examples/bri-3/components/identity/bpiAccount/bpiAccount.ts:9]: @skosito ideally the constructor caller function must be having the nonce increment logic. So I think it is fine to have the constructor initializer

**Comment by Manik-Jain** [comment]
> I concur with previous comments and approve pending suggested changes of @Therecanbeonlyone1969

+1

**Review by ognjenkurtic — APPROVED** [review]

**Review by Kasshern — APPROVED** [review]

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM

**Pull request merged**
